docs/spec.md

```diff
@@ -2828,6 +2829,43 @@ behavior is undefined. More formally, for all `id < jd` from `indices(result)`,
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.get_dimension_size
+
+### Semantics
+
+Returns the size of the given `dimension` of the `operand`.
+
+### Inputs
+
+| Name      | Type                         |
+|-----------|------------------------------|
+| `operand` | tensor of any supported type |
+| `index`   | constant of type `si64`      |
+
+### Outputs
+
+| Name     | Type                                |
+|----------|-------------------------------------|
+| `result` | 0-dimensional tensor of type `si32` |
+
+### Constraints
+
+  * (C1) 0 $\le$ `dimension` $\lt$ rank of `operand`. [todo](https://github.com/openxla/stablehlo/issues/790)
+  * (C2) dim(operand, dimension) $\le$ std::numeric_limits<int32_t>::max(). [todo](https://github.com/openxla/stablehlo/issues/790)
```

Comment:
Do we know what is  special about the dimension size of this particular `dimension` axis that https://github.com/tensorflow/tensorflow/blob/963d7a175b0c03b0d7b03286c0d9eb7098c29e5e/tensorflow/compiler/xla/service/shape_inference.cc#L2486 has a dedicated check only for it but not for other axes.

Reply:
That is because the result type of get_dimension_size and operand type of set_dimension_size is i32.

Reply:
makes sense. Thanks!